### PR TITLE
Clone 403 response in platform/api

### PR DESCRIPTION
**Pull request by jerekshoe**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
Cloning the JSON response before `.json()` is called so that 403 responses that aren't related to access token expiration can be read properly. vets-api can return 403s for other reason besides access token expiration but those error responses  throw an error message on this line: https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/utilities/api/index.js#L134 because they have already had `.json()` called on them

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by asg5704 — CHANGES_REQUESTED** [review]
I think this should be fine since we check for the `status` first.

However, can you add a test case in `platform/utilities/tests/api/index.unit.spec.js` for a `403`?

**Comment by asg5704** [comment]
@jerekshoe Is there any more context on why the `.clone()` is explicitly needed?  Like are there errors, or something that's not working somewhere?

File: src/platform/utilities/tests/api/index.unit.spec.js [hunk 1]
```diff
@@ -73,6 +73,25 @@ describe('apiRequest', () => {
       expect(error).to.deep.equal(JSON.parse(expected.response.body));
     });
   });
+
+  it('should return JSON on (status: 403)', async () => {
```
Comment (jerekshoe) [inline src/platform/utilities/tests/api/index.unit.spec.js:77]: @asg5704 I think this is basically what you were asking for, let me know if this needs any alterations

**Review by asg5704 — APPROVED** [review]

**Comment by asg5704** [comment]
> Yeah we are having issues where some endpoints can return a 403, but when they do, we get an error on this line: https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/utilities/api/index.js#L134 saying that the body has already been read

Okay cool. Are the endpoints random or is there any correlation to them? The reason I ask is because I remember the `.clone()` causing an isssue but I can't remember exactly where (if it was in unit tests or production with the Pre Check-in. I mean luckily it won't propagate to production so I'm fine approving, just curious 😄

**Comment by jerekshoe** [comment]
> > Yeah we are having issues where some endpoints can return a 403, but when they do, we get an error on this line: https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/utilities/api/index.js#L134 saying that the body has already been read
> 
> Okay cool. Are the endpoints random or is there any correlation to them? The reason I ask is because I remember the `.clone()` causing an isssue but I can't remember exactly where (if it was in unit tests or production with the Pre Check-in. I mean luckily it won't propagate to production so I'm fine approving, just curious 😄

I was specifically having issues with one of our Cypress intercepts attempting to return a 403 and this tripping the issue with body having already been read. Specifically this test https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/claims-status/tests/e2e/08.claim-letters.cypress.spec.js#L109

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**